src/controllers/libraryController.ts

```diff
@@ -31,36 +32,35 @@ export class LibraryController implements Disposable {
         this.disposable.dispose();
     }
 
-    public async addLibraries(libraryGlobs?: string[]) {
-        if (!libraryGlobs) {
-            libraryGlobs = [];
-            const workspaceFolder: WorkspaceFolder | undefined = Utility.getDefaultWorkspaceFolder();
-            const isWindows = process.platform.indexOf("win") === 0;
-            const results: Uri[] | undefined = await window.showOpenDialog({
-                defaultUri: workspaceFolder && workspaceFolder.uri,
-                canSelectFiles: true,
-                canSelectFolders: isWindows ? false : true,
-                canSelectMany: true,
-                openLabel: isWindows ? "Select jar files" : "Select jar files or directories",
-                filters: { Library: ["jar"] },
-            });
-            if (!results) {
-                return;
-            }
-            libraryGlobs = await Promise.all(results.map(async (uri: Uri) => {
-                // keep the param: `includeWorkspaceFolder` to false here
-                // since the multi-root is not supported well for invisible projects
-                const uriPath = workspace.asRelativePath(uri, false);
-                return (await fse.stat(uri.fsPath)).isDirectory() ? `${uriPath}/**/*.jar` : uriPath;
-            }));
-        }
-
+    public async addLibraryGlobs(libraryGlobs: string[]) {
         const setting = Settings.referencedLibraries();
         setting.exclude = this.dedupAlreadyCoveredPattern(libraryGlobs, ...setting.exclude);
         setting.include = this.updatePatternArray(setting.include, ...libraryGlobs);
         Settings.updateReferencedLibraries(setting);
     }
 
+    public async addLibraries(canSelectFolders?: boolean) {
+        const workspaceFolder: WorkspaceFolder | undefined = Utility.getDefaultWorkspaceFolder();
+        const isMac = platform() === "darwin";
+        const results: Uri[] | undefined = await window.showOpenDialog({
+            defaultUri: workspaceFolder && workspaceFolder.uri,
+            canSelectFiles: !canSelectFolders,
+            canSelectFolders: canSelectFolders || isMac,
+            canSelectMany: true,
+            openLabel: canSelectFolders ? "Select Library Folders" : "Select Jar Libraries",
+            filters: canSelectFolders ? { Folders: ["*"] } : { "Jar Files": ["jar"] },
+        });
+        if (!results) {
+            return;
+        }
+        this.addLibraryGlobs(await Promise.all(results.map(async (uri: Uri) => {
+            // keep the param: `includeWorkspaceFolder` to false here
+            // since the multi-root is not supported well for invisible projects
+            const uriPath = workspace.asRelativePath(uri, false);
+            return canSelectFolders ? uriPath + "/**/*.jar" : uriPath;
```

Comment:
When it's on MacOS, and triggered from `addLibraries` (`canSelectFolders` is `undefined`), if the user selects a folder, then it will not append `/**/*.jar`.

SO maybe we still need the previous logic -- check if it is a folder or not and return different value accordingly.

Reply:
Good catch! I missed that since I don't have a mac to test on, but I'll add the old logic back.

Reply:
Fixed in 2197367 (although I can't test if it works, since I don't have a mac).